pkg/gardenlet/controller/seed/seed.go

```diff
@@ -40,19 +43,22 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/workqueue"
+	componentbaseconfig "k8s.io/component-base/config"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // Controller controls Seeds.
 type Controller struct {
 	k8sGardenClient        kubernetes.Interface
 	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
+	getSeedClient          getSeedClient
```

Comment:
What is the motivation for this? Can't we have just `k8sSeedClient kubernetes.Interface` and in the test pass a fake/mock client? Why it needs to be a func that provides a seedClient?

Reply:
This is used to mock the result of this function https://github.com/gardener/gardener/blob/master/pkg/operation/seed/seed.go#L116, which creates new kubernetes client. It seemed easier to provide it as a dependency.